src/lib/REST/Value/ContentTree/NodeExtendedInfo.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AdminUi\REST\Value\ContentTree;
+
+use Ibexa\Rest\Value as RestValue;
+
+/**
+ * @phpstan-type TRestrictions array{
+ *      hasAccess: bool,
+ *      restrictedContentTypeIds?: array<int>,
+ *      restrictedLanguageCodes?: array<string>,
+ * }
+ *
+ * @phpstan-type TPermissionRestrictions array{
+ *      create: TRestrictions,
+ *      edit: TRestrictions,
+ *      delete: TRestrictions,
+ *      hide: TRestrictions,
+ * }
+ */
+final class NodeExtendedInfo extends RestValue
+{
+    /** @phpstan-var TPermissionRestrictions|null */
+    private ?array $permissions;
+
+    /**
+     * @phpstan-param TPermissionRestrictions|null $permissions
+     */
+    public function __construct(
+        ?array $permissions = null,
```

Comment:
> Trailing comma in parameters list is only available since PHP 8.0 

Don't we have test coverage for it running on PHP 7.4? This should produce a fatal error.